manifest-editor/helpers/analyse.ts

```diff
@@ -0,0 +1,69 @@
+import { LanguageProperty } from "@iiif/presentation-2";
+import * as IIIFVault from "@iiif/vault";
+import { getValue } from "@iiif/vault-helpers";
+
+export type Image = {
+  id: string,
+  type: "Image",
+  format: string,
+  width: number,
+  height: number
+}
+
+export type Manifest = {
+  id: string,
+  type: "Manifest",
+  label: LanguageProperty
+}
+
+export type Collection = {
+  id: string,
+  type: "Collection",
+  label: LanguageProperty
+}
+
+
+const handleJSON = async (imageUrl: string) => {
+  let data: any = {};
+  if (!imageUrl) return;
+  try {
+    const response = await fetch(imageUrl);
+    if (!response.ok) {
+      throw new Error(response.statusText);
+    };
+    data = (await response.json()) as any;
+  } catch { };
+  return data;
+};
+
+export const analyse = async (url: string, expectedTypes?: Array<"Image" | "ImageService" | "Manifest" | "Collection">) => {
+  if (!url) return;
+  let data: any = {}
+
+  // Static images
+  if (url.includes(".jpg") || url.includes(".jpeg") || url.includes(".png")) {
+    data = {
+      id: url,
+      type: "Image",
```

Comment:
If we're in a browser environment, we don't need to put the image on the page (I don't think!) to get the height/width.
```js
const getImage = async (src) => {
    return new Promise((resolve, reject) => {
        const $img = document.createElement('img');
        $img.onload = () => resolve($img);
        $img.onerror = () => reject();
        $img.src = src;
        if ($img. complete) {
            resolve($img); // cached.
        }
    });
}

const image = await getImage(url);
image.naturalHeight;
image.naturalWidth;
```

Can also set:
```js
$img.crossOrigin = 'anonymous';
```
Maybe on a second try, if the first didn't work (or vice-versa)

Reply:
May also be required.
```js
const isNodeEnv = typeof window === 'undefined'
```

We could test that in a separate Jest test by setting:
```
/**
 * @jest-environment node
 */
```
(if the default has been set to `jsdom`)

Reply:
(Image code is untested!)

Reply:
@stephenwf thanks so much, this was a big help! 
Deployed and working here: https://deploy-preview-92--manifest-editor-testing.netlify.app/
(File -> Open)

Reply:
Nice, looks good 👍